src/rt/sched/schedulerthread.h

```diff
@@ -216,22 +216,21 @@ namespace verona::rt
       for (size_t i = 0; i < count; i++)
       {
         auto* cown = cowns[i];
-        auto bp = cown->backpressure.load(std::memory_order_relaxed);
+        auto bp = cown->bp_state.load(std::memory_order_relaxed);
         yield();
-        assert(!bp.muted());
+        assert(bp != BackpressureState::Muted);
 
         if (
-          (bp.unmutable()) || (state == ThreadState::PreScan) ||
-          (state == ThreadState::Scan) || (state == ThreadState::AllInScan))
+          (bp & BackpressureState::IsUnmutable) ||
```

Comment:
This is not clear from the enum declaration and is left as an implementation detail for anyone to either abuse or get it wrong. The lack of comments on the enum makes this issue worse.

Either hide the implementation on static methods and opaque enum (aka. unsigned) or make it explicit on the enum declaration that this is intentional and assert on places that write to `bp_state` that it cannot ever be `0b10`.

Reply:
I've added doc strings to the enum members. The `operator<<` on `BackpressureState` now aborts on the default branch, and the `check_backpressure_transition` would return false when either state is `IsUnmutable`.